core/ui/menu/qam/quick_access_menu.gd

```diff
@@ -1,45 +1,103 @@
 extends Control
 
+var state_machine := preload("res://assets/state/state_machines/global_state_machine.tres") as StateMachine
 const qam_state_machine := preload("res://assets/state/state_machines/qam_state_machine.tres")
 const OGUIButton := preload("res://core/ui/components/button.tscn")
 const transition_fade_in := preload("res://core/ui/components/transition_fade_in.tscn")
+
+var qam_button_state := preload("res://assets/state/states/qam_button_submenu.tres") as State
 var qam_state := preload("res://assets/state/states/quick_access_menu.tres") as State
 
 @onready var icon_bar: VBoxContainer = $MarginContainer/HBoxContainer/IconBar
 @onready var viewport: VBoxContainer = $MarginContainer/HBoxContainer/Viewport
+@onready var notifications_menu: HFlowContainer = $MarginContainer/HBoxContainer/Viewport/NotificationsMenu
+@onready var quick_settings_menu: Node = $MarginContainer/HBoxContainer/Viewport/QuickSettingsMenu
+@onready var last_child : Control = icon_bar.get_child(0)
 
 # Called when the node enters the scene tree for the first time.
 func _ready() -> void:
 	visible = false
 	qam_state.state_entered.connect(_on_state_entered)
 	qam_state.state_exited.connect(_on_state_exited)
+	
+	for child in icon_bar.get_children():
+		if not child is Control:
+			continue
+		child.focus_entered.connect(_on_child_focused.bind(child))
+		child.pressed.connect(_on_child_pressed)
+
+
+func _on_child_focused(child: Control) -> void:
+	last_child = child
+
+
+func _on_child_pressed() -> void:
+	if state_machine.current_state() == qam_button_state:
+		return
+	state_machine.push_state(qam_button_state)
```

Comment:
The `qam_button_state` shouldn't be used in the global state manager. We could probably get rid of that state alltogether. Just call `last_child.grab_focus()` here.

Reply:
I started with that. The global state manager will just close the QAM.

Reply:
Ah, yeah, `InputManager` is listening for back button presses and popping the global state. Thinking about it, maybe we should remove that logic from `InputManager` and let each menu decide how to handle back button presses.

Reply:
I think that change is outside the scope of this PR.

Reply:
Tracked in #19 to follow up on this

Reply:
#18